src/main/java/school/faang/photosync/GooglePhotosAutoUploader.java

```diff
@@ -0,0 +1,42 @@
+package school.faang.photosync;
+
+import java.util.ArrayList;
+import java.util.List;
+
+public class GooglePhotosAutoUploader {
+
+    Object lock = new Object();
+    List<String> photosToUpload = new ArrayList<>();
+
+    public void startAutoUpload() {
+        synchronized (lock) {
+            while (photosToUpload.isEmpty()) {
+                try {
+                    System.out.println("Waiting for new photos...");
+                    lock.wait();
+                } catch (InterruptedException e) {
+                    Thread.currentThread().interrupt();
+                    System.err.println("Thread was interrupted: " + e.getMessage());
+                }
+            }
+            uploadPhotos();
+        }
+    }
+
+    public void uploadPhotos() {
+        synchronized (lock) {
```

Comment:
когда метод startAutoUpload захватывает блокировку на объекте lock и вызывает wait(), он освобождает блокировку, позволяя другим потокам захватывать её.
когда lock.notify() вызывается в onNewPhotoAdded, поток startAutoUpload снова получает блокировку при пробуждении, и в этот момент он снова проверяет условие и может вызвать uploadPhotos.
с учетом того, что этот метод вызывается только из startAutoUpload блокировка тут избыточна.

Reply:
I think there needs to be a synchronized (lock) in the uploadPhotos method because I'm making changes to the photosToUpload list (photosToUpload.clear();)